src/entities/trade.test.ts

```diff
@@ -0,0 +1,661 @@
+
+      await expect(
+        Trade.fromRoutes(
+          [
+            { routev2, amount: amountv2 },
+            { routev2: route2v2, amount: amount2v2 },
+          ],
+          [{ routev3, amount: amountv3 }],
+          TradeType.EXACT_INPUT
+        )
+      ).rejects.toThrow('POOLS_DUPLICATED')
+    })
+
+    it('throws if routes have different inputs', async () => {
+      const routeOriginalV2 = new V2RouteSDK([pair_1_2], token1, token2)
+      const routev2 = new RouteV2(routeOriginalV2)
+      const amountv2 = CurrencyAmount.fromRawAmount(token1, JSBI.BigInt(100))
+
+      const routeOriginalV3 = new V3RouteSDK([pool_0_1, pool_1_2], token0, token2)
+      const routev3 = new RouteV3(routeOriginalV3)
+      const amountv3 = CurrencyAmount.fromRawAmount(token0, JSBI.BigInt(1000))
+
+      await expect(
+        Trade.fromRoutes([{ routev2, amount: amountv2 }], [{ routev3, amount: amountv3 }], TradeType.EXACT_INPUT)
+      ).rejects.toThrow('INPUT_CURRENCY_MATCH')
+    })
+
+    it('throws if routes have different outputs', async () => {
+      const routeOriginalV2 = new V2RouteSDK([pair_0_1], token0, token1)
+      const routev2 = new RouteV2(routeOriginalV2)
+      const amountv2 = CurrencyAmount.fromRawAmount(token0, JSBI.BigInt(100))
+
+      const routeOriginalV3 = new V3RouteSDK([pool_0_1, pool_1_2], token0, token2)
+      const routev3 = new RouteV3(routeOriginalV3)
+      const amountv3 = CurrencyAmount.fromRawAmount(token0, JSBI.BigInt(1000))
+
+      await expect(
+        Trade.fromRoutes([{ routev2, amount: amountv2 }], [{ routev3, amount: amountv3 }], TradeType.EXACT_INPUT)
+      ).rejects.toThrow('OUTPUT_CURRENCY_MATCH')
+    })
+  })
+  describe('#worstExecutionPrice', () => {
+    describe(' exact input swaps', () => {
+      const routev3 = new V3RouteSDK([pool_0_1, pool_1_2], token0, token2)
+      const route2v3 = new V3RouteSDK([pool_0_2], token0, token2)
+
+      const inputAmount = CurrencyAmount.fromRawAmount(token0, 100)
+      const outputAmount = CurrencyAmount.fromRawAmount(token2, 69)
+      const tradeType = TradeType.EXACT_INPUT
+
+      const exactInV3 = new Trade({
+        v2Routes: [],
+        v3Routes: [{ routev3, inputAmount, outputAmount }],
+        tradeType,
+      })
+
+      const exactInMultiRoute = new Trade({
+        v2Routes: [],
+        v3Routes: [
+          {
+            routev3,
+            inputAmount: CurrencyAmount.fromRawAmount(token0, 50),
+            outputAmount: CurrencyAmount.fromRawAmount(token2, 35),
+          },
+          {
+            routev3: route2v3,
+            inputAmount: CurrencyAmount.fromRawAmount(token0, 50),
+            outputAmount: CurrencyAmount.fromRawAmount(token2, 34),
+          },
+        ],
+        tradeType: TradeType.EXACT_INPUT,
+      })
+
+      it('throws if less than 0', () => {
+        expect(() => exactInV3.minimumAmountOut(new Percent(-1, 100))).toThrow('SLIPPAGE_TOLERANCE')
```

Comment:
Looks like we are testing minimumAmountOut here? Should it be worstExecutionPrice?